src/observableInterfaces.ts

```diff
@@ -95,8 +95,8 @@ export interface ListenerParams<T = any> {
 }
 export type ListenerFn<T = any> = (params: ListenerParams<T>) => void;
 
-type PrimitiveKeys<T> = Pick<T, { [K in keyof T]-?: T[K] extends Primitive ? K : never }[keyof T]>;
-type NonPrimitiveKeys<T> = Pick<T, { [K in keyof T]-?: T[K] extends Primitive ? never : K }[keyof T]>;
+type PrimitiveProps<T> = Pick<T, { [K in keyof T]-?: T[K] extends Primitive ? K : never }[keyof T]>;
+type NonPrimitiveProps<T> = Pick<T, { [K in keyof T]-?: T[K] extends Primitive ? never : K }[keyof T]>;
```

Comment:
I renamed `Keys` to `Props`, because the resulting type represents an object with properties (key-value pairs). `Keys` is therefore a bit misleading imo, but if you want I can rename them back : )

Reply:
You're right, that's better 👍